torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1795,6 +1815,31 @@ def coerce_to_expected_memory_format(x: torch.Tensor, memory_format: MemoryForma
     return restrided
 
 
+@contextlib.contextmanager
+def _disable_saved_tensors_hooks():
+    error_message = (
+        "Saved tensors hooks in aot_autograd were inlined in forward and backward graph."
+        "Changing hooks must recompile."
```

Comment:
mind explaining what's going on here? I'm not sure why we would raise an error in this context (if the user changes their saved_tensor_hooks across iterations, that should always manifest as a recompile, not a hard error, no?)